generated/nidigital/nidigital/session.py

```diff
@@ -1011,226 +1011,401 @@ def _get_attribute_vi_string(self, attribute):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
-        return [int(failure_count_ctype[i]) for i in range(buffer_size_ctype.value)]
+        return
 
     @ivi_synchronized
-    def get_pin_name(self, pin_index):
-        r'''get_pin_name
+    def create_source_waveform_serial(self, waveform_name, data_mapping, sample_width, bit_order):
+        r'''create_source_waveform_serial
 
         TBD
 
+        Tip:
+        This method requires repeated capabilities (pins). If called directly on the
+        nidigital.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        nidigital.Session repeated capabilities container, and calling this method on the result.:
+
+            session.pins[0,1].create_source_waveform_serial(waveform_name, data_mapping, sample_width, bit_order)
+
         Args:
-            pin_index (int):
+            waveform_name (str):
 
+            data_mapping (int):
 
-        Returns:
-            name (str):
+            sample_width (int):
+
+            bit_order (int):
 
         '''
         vi_ctype = _visatype.ViSession(self._vi)  # case S110
-        pin_index_ctype = _visatype.ViInt32(pin_index)  # case S150
-        name_buffer_size_ctype = _visatype.ViInt32()  # case S170
-        name_ctype = None  # case C050
-        error_code = self._library.niDigital_GetPinName(vi_ctype, pin_index_ctype, name_buffer_size_ctype, name_ctype)
-        errors.handle_error(self, error_code, ignore_warnings=True, is_error_handling=False)
-        name_buffer_size_ctype = _visatype.ViInt32(error_code)  # case S180
-        name_ctype = (_visatype.ViChar * name_buffer_size_ctype.value)()  # case C060
-        error_code = self._library.niDigital_GetPinName(vi_ctype, pin_index_ctype, name_buffer_size_ctype, name_ctype)
+        pin_list_ctype = ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case C010
+        waveform_name_ctype = ctypes.create_string_buffer(waveform_name.encode(self._encoding))  # case C020
+        data_mapping_ctype = _visatype.ViInt32(data_mapping)  # case S150
+        sample_width_ctype = _visatype.ViUInt32(sample_width)  # case S150
+        bit_order_ctype = _visatype.ViInt32(bit_order)  # case S150
+        error_code = self._library.niDigital_CreateSourceWaveformSerial(vi_ctype, pin_list_ctype, waveform_name_ctype, data_mapping_ctype, sample_width_ctype, bit_order_ctype)
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
-        return name_ctype.value.decode(self._encoding)
+        return
 
     @ivi_synchronized
-    def _get_pin_results_pin_information(self):
-        r'''_get_pin_results_pin_information
+    def fetch_history_ram_cycle_information(self, site, position, samples_to_read):
```

Comment:
Just realized that in #1111, `sites` is being made into a repeated capability. We need to choose one over the other.

Reply:
Will be handled in [this](https://github.com/ni/nimi-python/pull/1290/files#r387789387) PR.